components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/bottomsheet/BottomSheetView.kt

```diff
@@ -0,0 +1,170 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.windowInsetsBottomHeight
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.FilledIconButton
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButtonDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.Surface
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.launch
+import uk.gov.hmrc.components.compose.R
+import uk.gov.hmrc.components.compose.molecule.bottomsheet.BottomSheetView.SheetBottomTokens.DockedDragHandleColor
+import uk.gov.hmrc.components.compose.molecule.bottomsheet.BottomSheetView.SheetBottomTokens.DockedDragHandleHeight
+import uk.gov.hmrc.components.compose.molecule.bottomsheet.BottomSheetView.SheetBottomTokens.DockedDragHandleWidth
+import uk.gov.hmrc.components.compose.molecule.bottomsheet.BottomSheetView.SheetBottomTokens.DragHandleVerticalPadding
+import uk.gov.hmrc.components.compose.ui.theme.HmrcGrey2
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme.colors
+
+object BottomSheetView {
+
+    @OptIn(ExperimentalMaterial3Api::class)
+    @Suppress("LongMethod")
+    @Composable
+    operator fun invoke(
+        modifier: Modifier = Modifier,
+        showBottomSheet: Boolean,
+        skipPartiallyExpanded: Boolean = false,
+        enableFullScreenExpansion: Boolean = false,
+        topWindowInset: Dp = HmrcTheme.dimensions.hmrcSpacing48,
+        onDismissRequest: () -> Unit,
+        sheetContent: @Composable ColumnScope.() -> Unit,
+    ) {
+        val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = skipPartiallyExpanded)
+        val scope = rememberCoroutineScope()
+        val closeSheetContentDescription = stringResource(id = R.string.bottomSheetView_close_content_desc)
+
+        if (showBottomSheet) {
+            ModalBottomSheet(
+                onDismissRequest = onDismissRequest,
+                sheetState = sheetState,
+                containerColor = colors.hmrcWhiteBackground,
+                windowInsets = WindowInsets(top = topWindowInset),
```

Comment:
Yeah, I was having the same issue. I've added it as a parameter so we can try and provide the dynamic status bar height when we come to use the component in the main app, but 48.dp seems to be a good default otherwise.